File: NACA0012/postproc/stat.R
require(ggplot2)
data = read.table("fort.9",col.names = c("time","CFL","drag","lift","Fz"))
l = length(data$time)
n = 2000
j = round(seq(10,l,len=min(l,n)))
data = data[j,]
CFL = qplot(time,CFL,data=subset(data,CFL<1),geom="line")
t = 0.75*max(data$time)+0.25*min(data$time)
late = subset(data,time>t)
mdrag = mean(late$drag)
mlift = mean(late$lift)
adrag = mad(late$drag)
alift = mad(late$lift)
drag = qplot(time,drag,data=data,geom="line")
drag = drag+annotate("text",x=t,y=mdrag-2*adrag,label=paste("mean=",round(mdrag,3)," amp=",round(adrag,3)))
lift = qplot(time,lift,data=data,geom="line")
lift = lift+annotate("text",x=t,y=mlift-2*alift,label=paste("mean=",round(mlift,3)," amp=",round(alift,3)))

ppdf = function(plot,name){
     pdf(name,8,4)
     print(plot)
     dev.off()
}

ppdf(CFL,"CFL.pdf")
ppdf(drag,"drag.pdf")
ppdf(lift,"lift.pdf")

data = read.table("mgs.txt", col.names = c("itr","res"))

l = length(data$itr)
n = 2000
data$i = seq(1,l)
j = round(seq(1,l,len=min(l,n)))
#j = unique(c(which(data$itr>1),j))
data = data[j,]

itr = qplot(i,log(itr,2),data=data)+ylab(expression(log[2](iteration)))
res = qplot(i,log(res,10),data=data)+ylab(expression(log[10](residual)))
ppdf(itr,"itr.pdf")
ppdf(res,"res.pdf")

File: pulse/postproc/stat.py
#!/usr/bin/env python3
# ----------------------------------------- #
# stat.py
# ----------------------------------------- #
#
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from matplotlib.backends.backend_pdf import PdfPages
#
# read data and drop unwanted rows and columns
try:
    df = pd.read_csv('fort.9',delim_whitespace = True,
        names=["time","CFL","drag","lift","side","power","U"])
except FileNotFoundError:
    exit('stat: fort.9 not found')
df.drop(df.index[:3], inplace=True)

try:
    mg = pd.read_csv('fort.8',delim_whitespace = True,
        names=["itr","res0","res","inf"])[2:]
except FileNotFoundError:
    exit('stat: fort.8 not found')
#
# -- plot PDF pages
def str_rnd(num,d=4): return str(round(num,d))

def plot_hist(pdf,name,label):
    ax = df.plot(x='time',y=name,figsize=(8,4))
    plt.xlabel(r'$t/T$', fontsize=12)
    plt.ylabel(label, fontsize=12)
    mean,mad = df[name].mean(), 1.5748*df[name].mad()
    x1,x2,y1,y2 = plt.axis()
    mx,mn = min(y2,mean+3*mad),max(y1,mean-3*mad)
    plt.ylim([mn,mx])
    txt = 'mean='+str_rnd(mean)+', mad='+str_rnd(mad)
    plt.text(0.5,0.01,txt,transform=ax.transAxes)
    pdf.savefig()
    plt.close()

with PdfPages('history.pdf') as pdf:
    plot_hist(pdf,name='U',label=r'$U/U_j$')
    plot_hist(pdf,name='drag',label=r'$C_{Xp}$')
    plot_hist(pdf,name='lift',label=r'$C_{Yp}$')
    plot_hist(pdf,name='side',label=r'$C_{Zp}$')
    plot_hist(pdf,name='power',label=r'$C_{Pp}$')
    plot_hist(pdf,name='CFL',label=r'$\frac{\Delta t U}{\Delta x}$')

    mg.plot(y=['res0','res','inf'],figsize=(8,4))
    plt.yscale('log')
    pdf.savefig()

    mg.plot(y='itr',figsize=(8,4))
    pdf.savefig()
